### ISSUE TITLE
Compatibility of react-native-config with React Native 0.73+

### ISSUE THREAD

**Issue by dawidmatyjasik** [react-native-config/react-native-config#803]
Hello,

I am currently working on a project using React Native version 0.73.6 and I am `using react-native-config` to manage environment variables. Specifically, I am interested in ensuring that `productFlavors` are functioning correctly on Android.

I have a few questions regarding this setup:

- Is `react-native-config` fully compatible with React Native versions 0.73 and above?
- Has anyone successfully used `react-native-config` with Android `productFlavors` on these React Native versions?
- Are there any known issues or special configurations required to ensure that `react-native-config` works seamlessly with Android product flavors on React Native 0.73+?
- Is there any official documentation or guidance available for this specific use case?

Any insights or experiences shared would be greatly appreciated.

Thank you!

**Comment by maksimlya** [comment]
We are currently on react 0.73.5, and have no issues with this lib on android and product flavors. The setup is something like that:

In app/build.gradle flavors defined regular:

`productFlavors {
    DemoFlavor1 {
        applicationId "com.zzzzz.yyyyy"
        dimension "default"
        resValue "string", "app_name", "DemoFlavor1"
        versionCode 22222
        versionName "0.1.56"
    }
    DemoFlavor2 {
        applicationId "com.ccccc.vvvvv"
        dimension "default"
        resValue "string", "app_name", "DemoFlavor2"
        versionCode 33333
        versionName "0.2.54"
    }
}


    implementation project(':react-native-config')
`

And down in that file:

`project.ext.envConfigFiles = [
        "demoFlavor1debug":"../config/.env.DemoFlavor1",
        "demoFlavor1release":"../config/.env.DemoFlavor1",
        "demoFlavor2debug":"../config/.env.DemoFlavor2",
        "demoFlavor2release": "../config/.env.DemoFlavor2"
]
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"`


On iOS tho.... It is a pain and suffering due to caching mechanisms.

**Comment by RenanGarcia** [comment]
Here we started having problems on the javascript side after updating to version 0.73.6 of react native. During the android build we were able to normally use the variables defined in the .env files. We still haven't found the reason for this failure. If we find something we will report it here.

**Comment by luancurti** [comment]
Yes on both counts — answering the questions in order, with where things stand today (v1.7.1).

**RN 0.73+ compatibility.** Supported, with one version caveat: from **v1.6.0** the Android side uses `BaseReactPackage`, which React Native introduced in **0.74**. So on 0.73.6 specifically, pin **`react-native-config@1.5.10`**; on 0.74+ use the current release. New Architecture (TurboModules) support landed in 1.6.0 for iOS, Android and Windows. This is written up under [Android build error: cannot find symbol BaseReactPackage](https://github.com/react-native-config/react-native-config#android-build-error-cannot-find-symbol-basereactpackage).

**Product flavors.** Supported and documented — map each build variant to an env file with `project.ext.envConfigFiles`, declared *before* the `apply from` line: https://github.com/react-native-config/react-native-config#android-1

Two details about the matching that are easy to trip over:

- The variant name comes from the Gradle task name and is lowercased, and keys are lowercased too — so mixed-case keys are fine.
- Matching is `flavor.startsWith(key)` and the **first match wins**, so list the most specific keys first (`demoflavor1debug` before `demoflavor1`).
- `ENVFILE` in the environment takes priority over the whole map when it is set.

**One correction to the snippet above**, since people find this thread from search: on React Native 0.60+ do **not** add the manual-linking lines

```gradle
// settings.gradle - remove
include ':react-native-config'
project(':react-native-config').projectDir = ...

// app/build.gradle - remove
implementation project(':react-native-config')
```

Autolinking already registers the module, and on the New Architecture a manually linked module is never registered as a TurboModule — you get `TypeError: Cannot read property 'getConfig' of null`. Keep only the `apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"` line in `android/app/build.gradle`; that one is still required.

Also relevant if your flavors set their own `applicationId` (as in the example above): `BuildConfig` is generated in the module's `namespace`, not the `applicationId`. Since **v1.6.2** the library resolves that correctly on its own (#879). If your `BuildConfig` lives somewhere else entirely, point at it with `resValue "string", "build_config_package", "<your namespace>"`.

Closing as answered and locking the thread — if something here does not work in your setup, please open a new issue with your `build.gradle` and RN version.
